### ISSUE TITLE
[ENH] NOTEARS Algorithm 

### ISSUE THREAD

**Issue by satishmp** [pgmpy/pgmpy#1342]
### Subject of the issue
Requesting for adding a new structural learning algorithm (NOTEARS) for learning large networks.
This has been added to causalnex package.

Historically, structure learning has been a very hard problem. We are interested in looking for the optimal directed acyclic graph (DAGs) that describes the conditional dependencies between variables. However, the search space for this is combinatorial and scales super-exponentially with the number of nodes. NOTEARS algorithm cleverly introduces a new optimisation heuristic and approach to solving this problem, where the runtime for this is no longer exponential but scales cubically with the number of nodes.

References:
https://github.com/xunzheng/notears
https://causalnex.readthedocs.io/en/latest/05_resources/05_faq.html#why-notears-algorithm-over-other-structure-learning-methods

### Your environment
Enhancement request.

### Steps to reproduce
Enhancement Request.

### Expected behaviour
Enhancement request.

### Actual behaviour
Enhancement request.

**Comment by ankurankan** [comment]
@satishmp Yes, this is already on priority for me. I will work on it as soon as I get time. Or if you would like to contribute, you are most welcome.

**Comment by GunaPalanivel** [comment]
@ankurankan - Could you assign this to me :)
 Refer: [Findings](https://github.com/pgmpy/pgmpy/issues/2513#issuecomment-3968965229)

**Comment by Kshitizsikriwal** [comment]
Hi @ankurankan, I am looking to take up the NOTEARS implementation #1342 as a core part of my GSoC 2026 proposal. I have been studying Zheng et al. formulation and how libraries like causalNex leverage it for scalability.

Here is my proposed approach for integrating this into pgmpy. I'll formulate the structure learning as a continuous optimization problem minimizing least-squares loss subject to the trace-exponential acyclicity constraint. Since NOTEARS outputs a continuous weighted adjacency matrix, the estimator will include a w_threshold parameter in the fit() method to prune weak edges and reliably convert the continuous output back into a strict pgmpy.base.DAG object. The class will inherit from pgmpy.causal_discovery.base._BaseCausalDiscovery to ensure standard Scikit-learn style compatibility accepting pandas DataFrames.  Can you please assign this ?
